src/AppBundle/Controller/ComponentController.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+	namespace AppBundle\Controller;
+
+
+	use AppBundle\Entity\Component;
+	use AppBundle\Form\ComponentType;
+	use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+	use Symfony\Component\HttpFoundation\Request;
+	use Ob\HighchartsBundle\Highcharts\Highchart;
+
+
+	class ComponentController extends Controller
+	{
+		/***
+		* @param $dashboardId
+		* @return \Symfony\Component\HttpFoundation\RedirectResponse
+		*/
+		public function addComponentAction($dashboardId)
+		{
+			$component = new Component();
+
+			$em = $this->getDoctrine()
+					->getManager();
+
+			$component->setDashboard($em->getRepository('AppBundle:Dashboard')->find($dashboardId));
+
+			$em->persist($component);
+			$em->flush();
+
+
+
+			return $this->redirectToRoute('app_edit_component', array('componentId' => $component->getId()));
+			
+		}
+
+		/***
+		* @param Request $request
+		* @param $componentId
+		* @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+		*/
+		public function editComponentAction(Request $request, $componentId)
```

Comment:
You could use here ParamConverter http://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/converters.html